src/uwtools/drivers/ww3.py

```diff
@@ -0,0 +1,82 @@
+"""
+A library driver for ww3.
+"""
+
+from pathlib import Path
+from typing import List, Optional
+
+from iotaa import asset, task, tasks
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers.driver import Assets
+from uwtools.strings import STR
+
+
+class WaveWatchIII(Assets):
+    """
+    A library driver for ww3.
+    """
+
+    def __init__(
+        self,
+        config: Optional[Path] = None,
+        dry_run: bool = False,
+        batch: bool = False,
+        key_path: Optional[List[str]] = None,
+    ):
+        """
+        The driver.
+
+        :param config: Path to config file (read stdin if missing or None).
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        :param key_path: Keys leading through the config to the driver's configuration block.
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, key_path=key_path)
+
+    # Workflow tasks
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "ww3_shel.nml"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=self._driver_config["namelist"],
+            path=path,
+            schema=self._namelist_schema(),
+        )
```

Comment:
In line with other comments, and I think you're still working on it, but: This should probably be rendering a template instead of realizing a (namelist) config.

Reply:
Discussed offline - this should use `render` much like `mpas_base` does for `stream` files. Still some issues in tests, though, working on it